backend/coreapp/middleware.py

```diff
@@ -0,0 +1,68 @@
+from django.http.request import HttpRequest
+from django.utils.timezone import now
+from django.contrib import auth
+
+from .models import User, Profile
+import logging
+from rest_framework.request import Request as DRFRequest
+from typing import Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .github import GitHubUser
+
+class AnonymousUser(auth.models.AnonymousUser):
+    profile: Profile
+
+class Request(DRFRequest):
+    user: Union[User, AnonymousUser]
+    profile: Profile
+
+def disable_csrf(get_response):
+    def middleware(request: HttpRequest):
+        setattr(request, "_dont_enforce_csrf_checks", True)
+        return get_response(request)
+
+    return middleware
+
+def set_user_profile(get_response):
+    """
+    Makes sure that `request.profile` is always available, even for anonymous users.
+    """
+
+    def middleware(request: HttpRequest):
```

Comment:
I think if you change this `HttpRequest` to `Request`, then you can use plain `request.profile = profile` at the bottom of this fn instead of `setattr(...)`